patches/api/0329-Per-player-viewdistances.patch

```diff
@@ -0,0 +1,112 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Spottedleaf <Spottedleaf@users.noreply.github.com>
+Date: Fri, 29 Jan 2021 16:26:05 -0800
+Subject: [PATCH] Per player viewdistances
+
+Add per player no-tick, tick, and send view distances.
+
+Also add send view distance to World.
+
+diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
+index 8ae9198ba7fdb006dc420504a984627add20dbb5..44edfdeecfd57827bf73fe2348bdc056c3a46de8 100644
+--- a/src/main/java/org/bukkit/World.java
++++ b/src/main/java/org/bukkit/World.java
+@@ -3639,6 +3639,26 @@ public interface World extends PluginMessageRecipient, Metadatable, net.kyori.ad
+      * @param viewDistance view distance in [2, 32]
+      */
+     void setNoTickViewDistance(int viewDistance);
++
++    // Paper start - add view distances
++    /**
++     * Gets the sending view distance for this world.
++     * <p>
++     * Sending view distance is the view distance where chunks will load in for players in this world.
++     * </p>
++     * @return The sending view distance for this world.
++     */
++    public int getSendViewDistance();
++
++    /**
++     * Sets the sending view distance for this world.
++     * <p>
++     * Sending view distance is the view distance where chunks will load in for players in this world.
++     * </p>
++     * @param viewDistance view distance in [2, 32] or -1
++     */
++    public void setSendViewDistance(int viewDistance);
++    // Paper end - add view distances
+     // Paper end - view distance api
+ 
+     // Spigot start
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index 3ec1be36b90dbedb8631135555da4b69110e4791..5084ef3cb0884fb5c170859bc7be57a9fcba0a3c 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -1887,23 +1887,63 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+      * Gets the view distance for this player
+      *
+      * @return the player's view distance
+-     * @deprecated This is unimplemented and <i>will</i> throw an exception at runtime. The {@link org.bukkit.World World}-based methods still work.
++     * // Paper - implemented
+      * @see org.bukkit.World#getViewDistance()
+      * @see org.bukkit.World#getNoTickViewDistance()
+      */
+-    @Deprecated
++    //@Deprecated // Paper - implemented
+     public int getViewDistance();
+ 
+     /**
+      * Sets the view distance for this player
+      *
+      * @param viewDistance the player's view distance
+-     * @deprecated This is unimplemented and <i>will</i> throw an exception at runtime. The {@link org.bukkit.World World}-based methods still work.
++     * // Paper - implemented
```

Comment:
How does this get rendered?

Reply:
pretty sure that should be `<!-- Paper - implemented -->`, otherwise it'll render in the javadocs